fix: JsNumberLiteralExpression::as_number ignore trivia

TheEdward162 opened the pull request:
<!--
	Thanks for submitting a Pull Request! We appreciate you spending the time to work on these changes.
	Please provide enough information so that others can review your PR.
	Once created, your PR will be automatically labeled according to changed files.
	Learn more about contributing: https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md
-->

## Summary

`JsNumberLiteralExpression::as_number` wasn't ignoring trivia of the interal token, so it was almost always returning `None`.

## Test Plan

Added a doctest, it failed without the change and now it passes.

faultyserver commented:
Good catch! It looks like this also fixed an error in the snapshot test for `useNumericLiterals` https://github.com/biomejs/biome/actions/runs/7427689816/job/20215337416?pr=1447#step:5:1451. If you run `cargo insta review` it should show you the diff between the current and the new snapshot, and you can accept the new one through that interface (by pressing `a`). That will update the test and make it pass here on CI.

In general, you should also use `just ready` when you're ready to open a PR, which will run effectively all of the CI tests on your machine before pushing so you can catch this feedback earlier and update more quickly.

TheEdward162 commented:
Sorry, I don't really feel comfortable installing 8 command-line tools to run the testsuite. I've updated the snapshot test using the snapshot that got emitted using `cargo test`. I'll leave it to CI to check if everything works now.

Also I've rebased because I accidentally used the wrong author info.

ematipico commented on crates/biome_js_analyze/tests/specs/style/useNumericLiterals/invalid.js.snap:
```diff
@@ -1,5 +1,6 @@
 ---
 source: crates/biome_js_analyze/tests/spec_tests.rs
+assertion_line: 83
```
It seems you're running an old version of `insta`, because this line isn't needed anymore in new versions . Let's delete it :)
```suggestion
```
